apis/core-api/src/core/services/FanClubService.ts

```diff
@@ -161,6 +161,22 @@ class FanClubService extends BaseService {
   static async exists(id: DbIdentifier): ServiceByIdResultType<boolean> {
     return this.checkExistsById(db.FanClub, id);
   }
+
+  static async updateMembersCount(id: DbIdentifier) {
+    const fanClub = await db.FanClub.findByPk(id);
+    if (!fanClub) {
+      return;
+    }
+
+    const membersCount = await db.FanClubMember.count({
+      where: {
+        fanClubId: id,
+      },
+    });
+
+    fanClub.setDataValue('membersCount', membersCount);
+    await fanClub.save();
```

Comment:
does this function need to be used with accept invitation action ? if yes, this should need to wrap with transaction

Reply:
this method will be executed when:
1) someone is invited by the admin/owner
2) member requested to join the fan club
3) member rejected the invitation
4) admin/owner reject member request
5) someone leaves the fan club

Reply:
I think 1, 2, 3 should not use this method, instead, accept invitation may use this.
but the question was this function is not valid itself, because it should work after some actions on database, thats why I see the concern of wrapping transaction

Reply:
Do you mean to put `accept the invitation` and `calculate members count` actions into one transaction?
I think is an unnecessary step.